File: problems/13-figure-boxplot/starters/solution.R
# Problem 13 Starter (R)
# Create a publication-quality boxplot

library(ggplot2)
library(readr)

# Read weather data
weather <- read_csv('../02-csv-column-mean/input/weather.csv')

# TODO: Create boxplot using ggplot2
# - X-axis: station_id
# - Y-axis: temp_c
# - Use colorblind-friendly palette (e.g., scale_fill_manual with Okabe-Ito colors)
# - Add proper labels and title

# Suggested approach:
# p <- ggplot(weather, aes(x = station_id, y = temp_c, fill = station_id)) +
#   geom_boxplot() +
#   scale_fill_brewer(palette = 'Set2') +
#   labs(title = 'Daily Temperature Distribution by Station',
#        x = 'Station',
#        y = 'Temperature (°C)') +
#   theme_minimal() +
#   theme(legend.position = 'none')
#
# ggsave('boxplot.png', p, width = 89, height = 70, units = 'mm', dpi = 96)

print("TODO: Create boxplot")

File: problems/13-figure-boxplot/starters/solution.py
# Problem 13 Starter (Python)
# Create a publication-quality boxplot

import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns

# Read weather data
weather = pd.read_csv('../02-csv-column-mean/input/weather.csv')

# TODO: Create boxplot
# - X-axis: station_id
# - Y-axis: temp_c
# - Use colorblind-friendly palette
# - Add proper labels and title
# - Set figure size to ~89mm width (approximately 350px at 96dpi)

# Suggested approach:
# fig, ax = plt.subplots(figsize=(5, 4), dpi=96)
# sns.boxplot(data=weather, x='station_id', y='temp_c', palette='Set2', ax=ax)
# ax.set_xlabel('Station')
# ax.set_ylabel('Temperature (°C)')
# ax.set_title('Daily Temperature Distribution by Station')
# plt.tight_layout()
# plt.savefig('boxplot.png', dpi=96, bbox_inches='tight')
# plt.close()

print("TODO: Create boxplot")
